crypto/internal.h

```diff
@@ -696,7 +699,7 @@ OPENSSL_EXPORT void CRYPTO_free_ex_data(CRYPTO_EX_DATA_CLASS *ex_data_class,
 
 // Endianness conversions.
 
-#if defined(__GNUC__) && __GNUC__ >= 2
+#if defined(AWS_LC_BUILTIN_SWAP_SUPPORTED)
```

Comment:
I think that we don't need the additional `try_compile` logic for `bswap`. It should be possible to do it like this:
```suggestion
#if defined(_MSC_VER)
// use the _byteswap_ushort/ulong/uint64 here
#elif (__GNUC__ > 4) || (__GNUC__ == 4 && __GNUC_MINOR__ >= 8)
// use the __builtin_bswap16/32/64 here
#elif defined(__has_builtin) && __has_builtin(__builtin_bswap64)
// use the __builtin_bswap16/32/64 here
#else
// the portable implementation here
#endif
```